### PR TITLE
fix #2142 : migrating fragments to viewbinding

### PR DESCRIPTION
**Pull request by gururani-abhishek**
Fixes #2142 

light theme -> 

https://github.com/openMF/mifos-mobile/assets/59947871/af93e963-2125-485c-8ec2-050e9a583a79

dark theme -> 

https://github.com/openMF/mifos-mobile/assets/59947871/17539512-9eeb-43ba-adc0-3ac723d23f85

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt [hunk 1]
```diff
@@ -153,31 +114,64 @@ class LoanAccountsDetailFragment : BaseFragment(), LoanAccountsDetailView {
...
+        }
+
     }
 
     /**
      * Opens [SavingsMakeTransferFragment] to Make a payment for loan account with given
      * `loanId`
      */
-    @OnClick(R.id.btn_make_payment)
+//    @OnClick(R.id.btn_make_payment)
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt]: remove the comment here

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
just a minor change and everything else LGTM @gururani-abhishek

**Comment by gururani-abhishek** [comment]
Hey @jawidMuhammadi, please have a look at this whenever you'll find time.

**Comment by gururani-abhishek** [comment]
> just a minor change and everything else LGTM @gururani-abhishek

Thanks for the Review @PratyushSingh07, made the changes.

**Comment by gururani-abhishek** [comment]
@jawidMuhammadi please have a look at this whenever you find time.

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt [hunk 2]
```diff
@@ -110,73 +32,73 @@ class LoanAccountSummaryFragment : BaseFragment() {
...
 
     /**
      * Sets basic information about a Loan Account
      *
      * @param loanWithAssociations object containing details of each loan account,
      */
-    fun showLoanAccountsDetail(loanWithAssociations: LoanWithAssociations?) {
-        llLoanSummary?.visibility = View.VISIBLE
+    private fun showLoanAccountsDetail(loanWithAssociations: LoanWithAssociations?) {
+        binding.llLoanSummary.visibility = View.VISIBLE
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt]: You could use Kotlin's `with` and avoid using the `binding` variable many times inside the function.

```
with(binding){
  llLoanSummary.visibility = View.VISIBLE
  tvLoanProductName.text = loanWithAssociations?.loanProductName
 ....
}
```

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt (hunk 2 above)
Comment (gururani-abhishek) [reply app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt]: Thanks @jawidMuhammadi, got it. Made the changes.

Comment (jawidMuhammadi) [reply app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt]: @gururani-abhishek and @PratyushSingh07 whenever there is a code change request from someone on your PR, please don't mark it as resolved. Once it is done, ping the reviewer and let them mark it as resolved.

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt [hunk 3]
```diff
@@ -153,67 +114,94 @@ class LoanAccountsDetailFragment : BaseFragment(), LoanAccountsDetailView {
...
-        tvAccountNumberName?.text = loanWithAssociations?.accountNo
-        tvLoanTypeName?.text = loanWithAssociations?.loanType?.value
-        tvCurrencyName?.text = loanWithAssociations?.summary?.currency?.code
+        binding.tvAccountNumber.text = loanWithAssociations?.accountNo
+        binding.tvLoanType.text = loanWithAssociations?.loanType?.value
+        binding.tvCurrency.text = loanWithAssociations?.summary?.currency?.code
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt:158]: I believe you could use the `with` here as well, so in any place when you need a `binding` instance for more than 2 items I recommend you to use this function. Do the same in all other open PRs as well, please 😊 

cc @PratyushSingh07

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt (hunk 2 above)
Comment (gururani-abhishek) [reply app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountSummaryFragment.kt]: alright, got it.

File: app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt (hunk 3 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/LoanAccountsDetailFragment.kt:158]: sure @jawidMuhammadi !!

**Review by jawidMuhammadi — APPROVED** [review]

**Pull request merged**